From 00f65b78e6de520a4820702207ce098c6b073724 Mon Sep 17 00:00:00 2001
From: "Xiang, Haihao" <haihao.xiang@intel.com>
Date: Fri, 22 Feb 2013 13:56:04 +0800
Subject: [PATCH] Update the size of DMV buffer for H.264 decoding on IVB

It is at least width_in_mbs * align(height_in_mbs, 2) * 64.
Use width_in_mbs * (height_in_mbs + 1) * 64 in the driver.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=59050
Signed-off-by: Xiang, Haihao <haihao.xiang@intel.com>
---
 src/gen7_mfd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/gen7_mfd.c b/src/gen7_mfd.c
index 0fb7249..ff8ef11 100755
--- a/src/gen7_mfd.c
+++ b/src/gen7_mfd.c
@@ -191,7 +191,7 @@ gen7_mfd_init_avc_surface(VADriverContextP ctx,
     if (gen7_avc_surface->dmv_top == NULL) {
         gen7_avc_surface->dmv_top = dri_bo_alloc(i965->intel.bufmgr,
                                                  "direct mv w/r buffer",
-                                                 width_in_mbs * height_in_mbs * 64,
+                                                 width_in_mbs * (height_in_mbs + 1) * 64,
                                                  0x1000);
         assert(gen7_avc_surface->dmv_top);
     }
@@ -200,7 +200,7 @@ gen7_mfd_init_avc_surface(VADriverContextP ctx,
         gen7_avc_surface->dmv_bottom == NULL) {
         gen7_avc_surface->dmv_bottom = dri_bo_alloc(i965->intel.bufmgr,
                                                     "direct mv w/r buffer",
-                                                    width_in_mbs * height_in_mbs * 64,                                                    
+                                                    width_in_mbs * (height_in_mbs + 1) * 64,
                                                     0x1000);
         assert(gen7_avc_surface->dmv_bottom);
     }
-- 
1.8.1.3

